org.eclipse.scanning.event/src/org/eclipse/scanning/event/ConsumerImpl.java

```diff
@@ -416,86 +451,141 @@ private void terminateProcess(IConsumerProcess<U> process) throws EventException
 	public void run() throws EventException {
 		
 		startJobManager();
+		init();
+		
+		while(isActive()) {
+        	try {
+	            boolean ok = consume();
+	            if (!ok) break;
+	                     		
+        	} catch (Throwable ne) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Catch Exception instead of Throwable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1181)